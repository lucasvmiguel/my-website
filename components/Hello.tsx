interface HelloProps {
  name: string
}

const Hello = ({ name }: HelloProps) => {
  return (
    <>
      <p data-testid="hello-title">Hello {name}</p>
      <span>another nice message</span>
    </>
  )
}

export default Hello
