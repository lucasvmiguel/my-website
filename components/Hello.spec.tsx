import renderer from 'react-test-renderer'
import { render } from '@testing-library/react'

import Hello from './Hello'

it('renders Hello component correctly - snapshot', () => {
  const tree = renderer
    .create(<Hello name="Lucas"></Hello>)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders Hello component correctly - dom test', () => {
  const { getByTestId } = render(<Hello name="Lucas"></Hello>)
  const { textContent } = getByTestId('hello-title')

  expect(textContent).toBe('Hello Lucas')
})
