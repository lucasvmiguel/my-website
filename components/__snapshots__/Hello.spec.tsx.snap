// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Hello component correctly - snapshot 1`] = `
Array [
  <p
    data-testid="hello-title"
  >
    Hello 
    Lucas
  </p>,
  <span>
    another nice message
  </span>,
]
`;
